spec/acceptance/018_shield_spec.rb

```diff
@@ -0,0 +1,306 @@
+require 'spec_helper_acceptance'
+
+describe "elasticsearch shield" do
+
+  # Template manifest
+  let :base_manifest do <<-EOF
+class { 'elasticsearch' :
+  java_install => true,
+  manage_repo  => true,
+  repo_version => '#{test_settings['repo_version']}',
+}
+
+elasticsearch::plugin { 'elasticsearch/license/latest' :  }
+elasticsearch::plugin { 'elasticsearch/shield/latest' : }
+EOF
+  end
+
+  describe 'user authentication' do
+
+    describe 'single instance manifest' do
+
+      let :single_manifest do
+        base_manifest + <<EOF
+elasticsearch::instance { ['es-01'] :  }
+
+Elasticsearch::Plugin { instances => ['es-01'],  }
+
+elasticsearch::shield::user { '#{@user}':
+  password => '#{@user_password}',
+  roles    => ['admin'],
+}
+elasticsearch::shield::user { '#{@user}pwchange':
+  password => '#{@user_password}',
+  roles    => ['admin'],
+}
+EOF
+      end
+
+      it 'should apply cleanly' do
+        apply_manifest single_manifest, :catch_failures => true
+      end
+
+      it 'should be idempotent' do
+        expect(apply_manifest(
+          single_manifest, :catch_failures => true
```

Comment:
You can also do `:catch_changes => true`, so you don't have to check the exit code.